Correct telemetry for Prometheus doc (#2866)
istio/istio.io: content/docs/tasks/telemetry/querying-metrics/index.md

the exposed metrics. It provides a mechanism for persistent storage and querying
of Istio metrics.

The configured Prometheus add-on scrapes the following endpoints:

> 1. `istio-telemetry.istio-system:42422`: The `istio-mesh` job returns all Mixer-generated mesh.

Reviewer: mesh metrics

Author: done

the exposed metrics. It provides a mechanism for persistent storage and querying
of Istio metrics.

The configured Prometheus add-on scrapes the following endpoints:

> 1. *istio-mesh* (`istio-telemetry.istio-system:42422`): all Mixer-generated mesh

Reviewer 2: Why are these not capitalized and separated. Is that the actual value of the endpoint? That case seems unlikely since I see the actual code in the parenthesis. Please capitalize appropriately and write them as separate words. Please use verbs and prepositions to make the actions explicit:
```suggestion
1. **Istio mesh** addressable at `istio-telemetry.istio-system:42422`: Returns all Mixer-generated mesh
```

Reviewer 2: Then, please refer to them as jobs and link to that page in your content. For example: The `istio-mesh` job ...

Author: The page is used to tell the user how to verify the prometheus configurations. I do not think it is proper to link that here. How about change to - `istio-telemetry.istio-system:42422`: The `istio-mesh` job returns all Mixer-generated mesh

   metrics.

> 1. *istio-telemetry* (`istio-telemetry.istio-system:9093`): all Mixer-specific metrics. Used

Reviewer 2: See my comment on line 96:
```suggestion
1. **Istio telemetry** addressable at `istio-telemetry.istio-system:9093`: Returns all Mixer-specific metrics. Use this endpoint
```

   to monitor Mixer itself.

> 1. *envoy-stats* (`istio-proxy:15090`): raw stats generated by Envoy. Prometheus is configured to look for pods with the `envoy-prom` endpoint exposed. The add-on configuration filters out a large number of envoy metrics during collection in an attempt to limit the scale of data processed by the add-on.

Reviewer 2: See my comment on line 96:
```suggestion
1. Envoy stats addressable at `istio-proxy:15090`: Returns the raw stats generated by Envoy. Prometheus is configured to look for pods with the `envoy-prom` endpoint exposed. The add-on configuration filters out a large number of envoy metrics during collection in an attempt to limit the scale of data the add-on processes.
```

> 1. *pilot* (`istio-pilot.istio-system:9093`): all pilot metrics.

Reviewer 2: ```suggestion
1. **Pilot** addressable at`istio-pilot.istio-system:9093`: Returns all the Pilot-generated metrics.
```

> 1. *galley* (`istio-galley.istio-system:9093`): all galley metrics.

Reviewer 2: ```suggestion
1. **Galley** addressable at `istio-galley.istio-system:9093`: Returns all the Galley-generated metrics.
```

> 1. *istio-policy* (`istio-policy.istio-system:9093`): all policy metrics.

Reviewer 2: ```suggestion
1. **Istio policy** addressable at `istio-policy.istio-system:9093`: Returns all policy-related metrics.
```

The configured Prometheus add-on scrapes three endpoints:

1. *istio-mesh* (`istio-telemetry.istio-system:42422`): all Mixer-generated mesh
   metrics.
1. *mixer* (`istio-telemetry.istio-system:9093`): all Mixer-specific metrics. Used
   to monitor Mixer itself.

> 1. *envoy* (`istio-proxy:15090`): raw stats generated by Envoy.

Reviewer 3: while we are here, can we also add entries for pilot, galley, and istio-policy  (and update the "The configured Prometheus add-on scrapes three endpoints:" to read "The configured Prometheus add-on scrapes the following endpoints:") ?

Author: added.

> Prometheus configures to look for pods with the endpoint exposed and filters out a large number of envoy metrics based on scraping configuration. For more on querying Prometheus, please read their [querying

Reviewer 3: nit: "Prometheus *is* configured to look for pods with the `envoy-prom` endpoint exposed. The add-on configuration filters out a large number of envoy metrics during collection in an attempt to limit the scale of data processed by the add-on."

Author: updated
